spatialmath/capsule.go

```diff
@@ -373,3 +374,140 @@ func separatingAxisTest1D(positionDelta, capVec *r3.Vector, plane r3.Vector, hal
 	sum -= math.Abs(capVec.Dot(plane))
 	return sum
 }
+
+// CapsuleIntersectionWithPlane calculates the intersection of a geometry with a plane and returns
+// a list of points along the surface of the geometry at the points of intersection.
+// It returns an error if the geometry type is unsupported or if points cannot be computed.
+// The points returned are in order, in frame of the capsule's parent, and follow the right hand rule around the plane normal.
+func CapsuleIntersectionWithPlane(g Geometry, planeNormal, planePoint r3.Vector, numPoints int) ([]r3.Vector, error) {
+	c, ok := g.(*capsule)
```

Comment:
[nit] can use `utils.AssertType`

Reply:
Is there really a benefit to this? It's the same number of LoC, but now with a function call abstraction.

Reply:
I like it personally since it cuts down on needing to write similar errors in a ton of places. I've been trying to promote it where applicable for better uniformity. 

The only knock on it I see is the overhead of an additional function call?

Reply:
I'm less worried about the computational overhead of the function call, and more worried about the overhead from a code readability standpoint. Instead of doing the thing in one line right there, we abstract it away in a function call to another package which, if there is unexpected behavior going on, requires way more effort to track down and check. It's a higher mental load to ensure code correctness, without a line count reduction to make up for it.